enhancements/openshift-route-admission-policy.md

```diff
@@ -0,0 +1,133 @@
+
+## Summary
+
+See the [section][#route-admission-policies]
+
+## Motivation
+
+See the [section][#route-admission-policies]
+
+### Goals
+
+1. Define an API that allows the cluster administrator to enable sharing of hostnames across namespaces.
+
+### Non-Goals
+
+## Proposal
+
+Modify the existing `operator.openshift.io`'s custom resource [`IngressController`][ingress-controller-api] to include an enum field `RouteAdmissionPolicy` that can be set to `AllowInterNamespaceClaims` or `Strict`, where `Strict` is default.
+
+
+### User Stories
+
+#### Default Route Admission Policy
+
+The default behavior is `Strict` which match the current behavior and is a secure default.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+The openshift-ingress-operator will be able to use the enum `RouteAdmissionPolicy` to enable the environment variable `ROUTER_DISABLE_NAMESPACE_OWNERSHIP_CHECK` in the openshift-ingress router. This is enabled when `RouteAdmissionPolicy` is set to `AllowInterNamespaceClaims`.
+
+### Risks and Mitigations
+
+Addition of the new field requires no migration, as the default matches the current configuration.
+
+## Design Details
+
+### New API
+
+`RouteAdmissionPolicy` enum definition.
+
+```go
+// RouteAdmissionPolicy is an admission policy for allowing new route claims.
+type RouteAdmissionPolicy string
+
+const (
+    // AllowInterNamespaceClaimsRouteAdmissionPolicy allows routes to claim the same hostname across namespaces.
+    AllowInterNamespaceClaimsRouteAdmissionPolicy RouteAdmissionPolicy = "AllowInterNamespaceClaims"
+
+    // StrictRouteAdmissionPolicy does not allow routes to claim the same hostname across namespaces.
+    // This is the default behavior 
+    StrictRouteAdmissionPolicy RouteAdmissionPolicy = "Strict"
+)
+```
+
+### Migration of existing clusters
+
+All clusters with `empty` RouteAdmissionPolicy can continue running with the current default, without setting a value.
+
+#### Rollbacks
+
+In case of rollbacks where AllowInterNamespaceClaims has been used, administrators would have to move routes to a shared namespace.
+
+### Test Plan
+
+- Create two routes in two different namespaces with the same hostname, and ensure that they are not admitted by default (`empty` or `unset`) and `Strict`.
+
+- Create two routes in two different namespaces with the same hostname, and ensure that they are admitted when RouteAdmissionPolicy is set to `AllowInterNamespaceClaims`.
+
+#### Migration
+
+No migration should be required.
+
+### Graduation Criteria
+
+### Upgrade / Downgrade Strategy
+
+See the [rollback section][#rollbacks]
+
+### Version Skew Strategy
+
+The API defines `empty` equivalent to `Strict`, therefore all the clients will have to keep that check to remain backwards compatible.
+
+## Implementation History
+
+## Drawbacks
+
+This only implements one existing feature in the OpenShift router that existed in OpenShift 3.x, whereas there are many more configurations.
+
+## Alternatives
```

Comment:
Introducing a direct proxy of the old config will work, but I'm curious if there are other approaches which might not require new API. @smarterclayton and I have talked some about potential RBAC models, but it's not clear whether they would be a replacement or a supplement.

Any other novel ways of approaching the problem?